django_projects/locallibrary/catalog/views.py

```diff
@@ -16,3 +19,30 @@ def index(request):
 
     return render(request, 'index.html', context=context)
     
+class BookListView(generic.ListView):
+    model = Book
+    paginate_by = 10
+    def get_context_data(self, **kwargs):
+        context = super(BookListView, self).get_context_data(**kwargs)
+        context['some_data'] = 'This is just some book data'
+        return context
+
+class BookDetailView(generic.DetailView):
+    model = Book
+    def book_detail_view(request, primary_key):
+        book = get_object_or_404(Book, pk=primary_key)
+        return render(request, 'catalog/book_detail.html', context={'book': book})        
```

Comment:
1 cái là function view xây dựng trước đó, 1 cái là class view thêm vào sau, mình dùng 1 trong 2 cách e nha, nó tách riêng nhau, e có thể comment lại nếu ko dùng hoặc để vậy.
```suggestion
class BookDetailView(generic.DetailView):
    model = Book
    
def book_detail_view(request, primary_key):
    book = get_object_or_404(Book, pk=primary_key)
    return render(request, 'catalog/book_detail.html', context={'book': book})        
```

Reply:
Em sẽ dùng cách là class ạ